docs/react-18-and-react-native.md

```diff
@@ -0,0 +1,136 @@
+
+<Tabs groupId="android-language" defaultValue={constants.defaultAndroidLanguage} values={constants.androidLanguages}>
+
+<TabItem value="java">
+
+```diff
+public class MainActivity extends ReactActivity {
+
+  public static class MainActivityDelegate extends ReactActivityDelegate {
+    public MainActivityDelegate(ReactActivity activity, String mainComponentName) {
+      super(activity, mainComponentName);
+    }
+
+    @Override
+    protected ReactRootView createRootView() {
+      ReactRootView reactRootView = new ReactRootView(getContext());
+      // If you opted-in for the New Architecture, we enable the Fabric Renderer.
+      reactRootView.setIsFabric(BuildConfig.IS_NEW_ARCHITECTURE_ENABLED);
+      return reactRootView;
+    }
+
++   @Override
++   protected boolean isConcurrentRootEnabled() {
++     // If you opted-in for the New Architecture, we enable Concurrent Root (i.e. React 18).
++     // More on this on https://reactjs.org/blog/2022/03/29/react-v18.html
++     return BuildConfig.IS_NEW_ARCHITECTURE_ENABLED;
++   }
+  }
+}
+```
+
+</TabItem>
+
+<TabItem value="kotlin">
+
+```diff
+class MainActivity : ReactActivity() {
+
+    open class MainActivityDelegate(activity: ReactActivity?, mainComponentName: String?) : ReactActivityDelegate(activity, mainComponentName) {
+        override fun createRootView(): ReactRootView = ReactRootView(context).apply {
+            // If you opted-in for the New Architecture, we enable the Fabric Renderer.
+            setIsFabric(BuildConfig.IS_NEW_ARCHITECTURE_ENABLED)
+        }
+
++       // If you opted-in for the New Architecture, we enable Concurrent Root (i.e. React 18).
++       // More on this on https://reactjs.org/blog/2022/03/29/react-v18.html
++       override fun isConcurrentRootEnabled() = BuildConfig.IS_NEW_ARCHITECTURE_ENABLED
+    }
+}
+```
+
+</TabItem>
+</Tabs>
+
+### Enabling Concurrent React on iOS
+
+On iOS, you'll have access to the `concurrentRootEnabled` method on your `AppDelegate.mm` file.
+
+```objc
+/// This method controls whether the `concurrentRoot`feature of React18 is turned on or off.
+///
+/// @see: https://reactjs.org/blog/2022/03/29/react-v18.html
+/// @note: This requires to be rendering on Fabric (i.e. on the New Architecture).
+/// @return: `true` if the `concurrentRoot` feture is enabled. Otherwise, it returns `false`.
+- (BOOL)concurrentRootEnabled
+{
+  // Switch this bool to turn on and off the concurrent root
+  return true;
+}
+```
+
+### Users on React Native 0.69 not yet migrated to the New Architecture
+
+Please note that users on React Native 0.69, but still on the Old Architecture won't benefit from the new React 18 features.
+
+While you can still override the `isConcurrentRootEnabled` method, that value will have no effect on your rendering.
+You will still be using the old React rendered, you won't benefit of any concurrent features, and you won't be able to use the new
+`startTransition`/`useTransition` APIs.
```

Comment:
I'll rephrase this with something more assertive, perhaps even replacing the section with a `caution box`

```suggestion
:::caution

Users on the Old Architecture won't benefit from the new React 18 features, despite being on React Native 0.69.

The `isConcurrentRootEnabled` methods are ignored when running on the new architecture.
:::
```

I think that the box will catch more the attention of the user.
What do you think?

Reply:
How do you feel about the current wording?

```
### Users on React Native 0.69 not yet migrated to the New Architecture

Note: Users on React Native 0.69, but still on the Old Architecture will still use React 17 mode even if React 18 is installed in the `package.json` file.

Overriding the `isConcurrentRootEnabled` method will have no effect on your app.
```